Patch by Stuart Shelton for IRIX 32-bits support (using MIPSpro)

http://bugs.gentoo.org/show_bug.cgi?id=267105

--- src/common.h
+++ src/common.h
@@ -39,12 +39,10 @@
 
 #if (SIZEOF_LONG == 8)
 #	define	SF_PLATFORM_S64(x)		x##l
-#elif COMPILER_IS_GCC
-#	define	SF_PLATFORM_S64(x)		x##ll
 #elif OS_IS_WIN32
 #	define	SF_PLATFORM_S64(x)		x##I64
 #else
-#	error "Don't know how to define a 64 bit integer constant."
+#	define	SF_PLATFORM_S64(x)		x##ll
 #endif
 
 
--- tests/utils.h
+++ tests/utils.h
@@ -29,7 +29,9 @@
 extern "C" {
 #endif	/* __cplusplus */
 
+#ifndef __cplusplus
 #include <stdint.h>
+#endif
 #include <stdarg.h>
 
 #define SF_COUNT_TO_LONG(x)	((long) (x))
